lib/server/controllers/folder_controller.rb

```diff
@@ -112,4 +100,30 @@ def rename
 
     success_json(folders: [ folder.to_hash ])
   end
+
+  protected
+
+  def mkdir_p(bucket, folder_path, project_name, author)
+    existing_folders = Metis::Folder.from_path(bucket, folder_path, allow_partial_match: true)
+    folder_names = folder_path.split(%r!/!)
+
+    folder_names.inject([]) do |parents, folder_name|
+      existing = existing_folders.shift
+      next (parents << existing) unless existing.nil?
+
+
+      if Metis::File.exists?(folder_name, bucket, parents.last)
+        raise Etna::BadRequest, "Cannot overwrite existing file" if parents.length == folder_names.length - 1
+        raise Etna::BadRequest, "Invalid folder: \"#{folder_name}\""
```

Comment:
Why these two raises?

Reply:
Hmm - because you are no longer using "require_folder" you must replace its behavior here. I think in this context the previous behavior no longer applies - before, you were assuming the folder tree (i.e, the path) exists as a set of valid folders; it is immaterial whether there is a competing file or not, because we are not attempting to create the missing folders, we are only attempting to add a new leaf folder to the tree; in this case when it fails we should say "Invalid folder". However, here we ARE attempting to create the intervening folder. If a file exists, the user should be warned the same way; so we only need the first raise in all cases.

Reply:
Understood, will change, thank you!